R/read-data.R

```diff
@@ -185,3 +193,8 @@ read_area_merged <- function(file) {
 read_csv_partial_cols <- function(...){
   suppress_one_warning(readr::read_csv(...), "The following named parsers don't match the column names")
 }
+
+drop_na_rows <- function(x) {
+  na_rows <- apply(is.na(x) | x == "", 1, all)
```

Comment:
Looking at the way this is used x should never be "" here as `readr::read_csv` makes any empty string an NA. I think not a problem to leave it as is though.

Reply:
I noticed that when I was writing the test.  This behaviour mimics hintr: https://github.com/mrc-ide/hintr/blob/master/R/read_data.R#L24   

Is there any other reason that hintr checks the "" case?

Reply:
Yeah - in hintr we're using `read.csv` from R utils instead of `readr::read_csv`. The utils `read.csv`  will use "" if it thinks the column type is a character.

```
t <- tempfile()
writeLines(c("string_col,number_col", "testing,2", ","), t)
read.csv(t)
```

Gives
```
  string_col number_col
1    testing          2
2                    NA
```